src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml

```diff
@@ -114,6 +114,24 @@ services:
       - '@=service("prestashop.adapter.shop.context").getContextListShopIDUsingCustomerSharingSettings()'
     public: true
 
+  prestashop.core.grid.query_builder.customer_bought_product:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Query\CustomerBoughtProductQueryBuilder'
+    parent: 'prestashop.core.grid.abstract_query_builder'
+    arguments:
+      - '@prestashop.core.query.doctrine_search_criteria_applicator'
+      - "@=service('prestashop.adapter.legacy.context').getContext().language.id"
+      - '@=service("prestashop.adapter.shop.context").getContextListShopIDUsingCustomerSharingSettings()'
+    public: true
```

Comment:
For all services
```suggestion
    public: false
```

Reply:
@FabienPapet is it possible to decorate/override a service marked as `public: false`?

Reply:
yes